app/grandchallenge/core/static/js/refresh_sidebar.mjs

```diff
@@ -44,5 +44,5 @@ $(document).ready(function() {
     });
 
     activateLocation();
-    document.dispatchEvent(new Event("vega.render"));
+    window.dispatchEvent(new Event('resize'))
```

Comment:
Emitting a resize event on document load is really hacky. There are going to be other listeners to this event which will now fire completely unnecessarily.

Reply:
Actually, this is the same approach in place to make vega charts render correctly in the logs tab ([here](https://github.com/comic/grand-challenge.org/blob/main/app/grandchallenge/algorithms/static/algorithms/js/display_result_text.mjs)), however not on page loading but on tab showing, and the similar issue raised by others [here](https://talk.observablehq.com/t/embedding-vega-lite-charts-that-have-width-value-set-to-container-leads-to-null-width-until-window-is-resized).

I explained in the PR #3460 why it is happening and this behaviour appear differently on different browsers since the issue was initially reported on Safari and I found it on Brave too. Do you suggest a recommended way of tackling this issue?

Reply:
The other suggestion I had is to trigger this event only when rendering the vega-lite-chart  ([here](https://github.com/comic/grand-challenge.org/pull/3460/commits/37730ee6c7fae0376e560fc614bf5727103f0584))

Reply:
It's a hack there too. If I knew the solution I would have implemented it. You need to understand why the problem is happening yourself, and then address it in a direct way.